Allure.XUnit/AllureXunitHelper.cs

```diff
@@ -177,7 +177,7 @@ private static void AddDistinct(this List<Label> labels, string labelName, strin
 
         private static void UpdateTestDataFromAttributes(TestResult testResult, ITestCase testCase)
         {
-            var classAttributes = testCase.TestMethod.TestClass.Class.GetCustomAttributes(typeof(IAllureInfo));
+            var classAttributes = GetCustomAttributesRecursive(testCase.TestMethod.TestClass.Class, typeof(IAllureInfo));
```

Comment:
No need for `GetCustomAttributesRecursive` method.

`GetCustomAttributes` has an overload for getting attributes with inheritance:
https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=netcore-2.0#system-reflection-memberinfo-getcustomattributes(system-type-system-boolean)